pkg/util/cryptomaterial/certchains/signers.go

```diff
@@ -17,50 +17,194 @@ import (
 	"github.com/openshift/microshift/pkg/util/cryptomaterial"
 )
 
-type CertificateSigner struct {
-	signerName         string
-	signerConfig       *crypto.CA
-	signerDir          string
-	signerValidityDays int
+type CSRInfo interface{ GetMeta() CSRMeta }
 
-	subCAs             map[string]*CertificateSigner
-	signedCertificates map[string]*signedCertificateInfo
+type CSRMeta struct {
+	Name         string
+	ValidityDays int
 }
 
-type signedCertificateInfo struct {
-	certDir   string
-	tlsConfig *crypto.TLSCertificateConfig
+type CertificateRequestInfo interface {
```

Comment:
why do you need CertificateRequestInfo and CSRInfo? seem pretty the same, and the former is not been used.

Reply:
That must be a remnant of the past, I'll remove it. Good catch :slightly_smiling_face: